sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,467 @@
+executed asynchronously from the caller.  A single object of this class can map
+naturally to a single hardware instance on spatial targets.
+
+NOTE: In this document, we use `task_sequence` to indicate the proposed
+`sycl::ext::intel::experimental::task_sequence`
+
+=== Examples
+
+Here's an example that shows how to define and call two task sequences.
+
+```c++
+using namespace sycl::ext::intel::experimental;
+
+// Return the sum of elements in 'v', from index 's' to 's+sz'
+int vectorSum(float *v, size_t s, size_t sz) {
+  ...
+}
+
+{
+  sycl::queue q{...};
+
+  constexpr int kCount = ...;
+  std::vector<float> in(kCount);
+  float* out = ...;
+  ...
+
+  q.single_task([=] {
+    task_sequence<vectorSum> firstHalf;
+    task_sequence<vectorSum> secondHalf;
+    constexpr int halfCount = kCount/2;
+    firstHalf.async(in.data(), 0, halfCount);
+    secondHalf.async(in.data(), halfCount, halfCount);
+    *out = firstHalf.get() + secondHalf.get();
+  }).wait();
+  ...
+}
+```
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_FPGA_TASK_SEQUENCE` to one of the values defined in the
+table below.  Applications can test for the existence of this macro to determine
+if the implementation supports this feature, or applications can test the
+macro's value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+== Definitions
+
+== `task_sequence` class
+
+[source,c++,linenums]
+----
+namespace sycl::ext::intel::experimental {
+
+template<auto &f> struct task_sequence_checker : std::false_type {};
+
+template<typename ReturnT, typename... ArgsT, ReturnT (&f)(ArgsT...)>
+struct task_sequence_checker<f> : std::true_type {};
+
+template<auto& f, class propertiesT =
+  decltype(oneapi::experimental::properties{})>
+class task_sequence{ static_assert(task_sequence_checker<f>::value); };
+
+template <typename ReturnT, typename... ArgsT, ReturnT (&f)(ArgsT...),
+          class propertiesT>
+class task_sequence<f, propertiesT> {
+
+public:
+  task_sequence(const task_sequence &) = delete;
+  task_sequence &operator=(const task_sequence &) = delete;
+  task_sequence(task_sequence &&) = delete;
+  task_sequence &operator=(task_sequence &&) = delete;
+
+  task_sequence();
+
+  task_sequence(const propertiesT &);
+
+  void async(ArgsT... args);
+
+  ReturnT get();
+
+  template <typename propertyT>
+  static constexpr bool has_property();
+
+  template <typename propertyT>
+  static constexpr auto get_property();
+
+  ~task_sequence();
+};
+
+} // namespace experimental
+} // namespace intel
+} // namespace ext
+} // namespace sycl
```

Comment:
I think you just want one closing brace here.

Reply:
fixed